### PR TITLE
feat(#2506): versionize annotation

### PR DESCRIPTION
**Pull request by maxonfjvipon**
Ref: #2506 

Created `@Versionize` annotation and attached it to almost all classes in `eo-runtime`

---

## PR-Codex overview
This PR focuses on adding the `@Versionized` annotation to various classes and interfaces in the codebase. 

### Detailed summary
- Added `@Versionized` annotation to `Expr`, `Term`, `Main`, `Data`, `Indented`, `JavaPath`, `Param`, `CachedPhi`, `PhOnce`, `Universe`, `Dataized`, `Attr`, `AtFixed`, `AtNamed`, `PhNamed`, `AtAbsent`, `AtLogged`, `PhCopy`, `PhPackage`, `PhWith`, `AtLocated`, `AtOnce`, `PhSafe`, `AtSafe`, `PhConst`, `PhUnvar`, `AtCage`, `BytesOf`, `PhFake`, `PhImmovable`, `PhMethod`, `AtSimple`, `AtVararg`, `ExFailure`, `ExFlow`, `PhLogged`, `ExUnset`, `PhLocated`, `AtFree`, `AtComposite`, `AtPhiSensitive`, `ExprReduce`, `ExReadOnly`, `Bytes`, `ExAbstract`, `UniverseDefault`, `Volatile`, `XmirObject`, `AtConst`, `ExInterrupted`, `UniverseSafe`, `ExNative`, `Heaps`, `Input`, `Ram`, `PhDefault`, `ExReduceBytes`, `EOram$EOram_slice$EOφ`, `EOseq`, `EOas_phi`, `EOmemory`, `AtMemoized`, `EOerror`, `EOint$EOgt`, `EObool$EOif`, `EObool$EOor`, `EOstdout`, `EObytes$EOeq`

> The following files were skipped due to too many changes: `eo-runtime/src/main/java/EOorg/EOeolang/EObytes$EOeq.java`, `/eo-runtime/src/main/java/EOorg/EOeolang/EOio/EOstdin$EO\317\206.java"`, `eo-runtime/src/main/java/EOorg/EOeolang/EObool$EOand.java`, `eo-runtime/src/main/java/EOorg/EOeolang/EOfloat$EOgt.java`, `eo-runtime/src/main/java/EOorg/EOeolang/EOtuple$EOat.java`, `eo-runtime/src/main/java/EOorg/EOeolang/EObytes$EOor.java`, `eo-runtime/src/main/java/EOorg/EOeolang/EObytes$EOand.java`, `eo-runtime/src/main/java/EOorg/EOeolang/EObytes$EOnot.java`, `eo-runtime/src/main/java/EOorg/EOeolang/EObytes$EOxor.java`, `eo-runtime/src/main/java/EOorg/EOeolang/EOint$EOtimes.java`, `eo-runtime/src/main/java/EOorg/EOeolang/EOfloat$EOdiv.java`, `eo-runtime/src/main/java/EOorg/EOeolang/EOram$EOslice.java`, `eo-runtime/src/main/java/EOorg/EOeolang/EObool$EOwhile.java`, `eo-runtime/src/main/java/EOorg/EOeolang/EObytes$EOsize.java`, `eo-runtime/src/main/java/EOorg/EOeolang/EOram$EOwrite.java`, `eo-runtime/src/main/java/EOorg/EOeolang/EOtuple$EOwith.java`, `eo-runtime/src/main/java/EOorg/EOeolang/EOfloat$EOplus.java`, `eo-runtime/src/main/java/EOorg/EOeolang/EObytes$EOslice.java`, `eo-runtime/src/main/java/EOorg/EOeolang/EOfloat$EOtimes.java`, `eo-runtime/src/main/java/EOorg/EOeolang/EObytes$EOright.java`, `eo-runtime/src/main/java/EOorg/EOeolang/EOio/EOstdin$EOnext_line.java`, `eo-runtime/src/main/java/EOorg/EOeolang/EOtuple$EOlength.java`, `eo-runtime/src/main/java/EOorg/EOeolang/EObytes$EOas_int.java`, `eo-runtime/src/main/java/EOorg/EOeolang/EObytes$EOconcat.java`, `eo-runtime/src/main/java/EOorg/EOeolang/EOstring$EOslice.java`, `eo-runtime/src/main/java/EOorg/EOeolang/EOstring$EOlength.java`, `eo-runtime/src/main/java/EOorg/EOeolang/EOcage.java`, `eo-runtime/src/main/java/EOorg/EOeolang/EOstring$EOas_bytes.java`, `eo-runtime/src/main/java/EOorg/EOeolang/EObytes$EOas_string.java`, `eo-runtime/src/main/java/EOorg/EOeolang/EOrust.java`, `eo-runtime/src/main/java/EOorg/EOeolang/EOram$EOram_slice$EOwrite.java`, `eo-runtime/src/main/java/EOorg/EOeolang/EOheap$EOpointer$EOblock.java`, `eo-runtime/src/main/java/EOorg/EOeolang/EOtry.java`, `eo-runtime/src/main/java/EOorg/EOeolang/EOgoto.java`, `eo-runtime/src/main/java/EOorg/EOeolang/EOint$EOas_bytes.java`, `eo-runtime/src/main/java/EOorg/EOeolang/EOfloat$EOas_bytes.java`, `eo-runtime/src/main/java/EOorg/EOeolang/EObytes$EOas_float.java`, `eo-runtime/src/main/java/EOorg/EOeolang/EOint$EOplus.java`, `eo-runtime/src/main/java/EOorg/EOeolang/EOint$EOdiv.java`, `eo-runtime/src/main/java/org/eolang/Versionized.java`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by maxonfjvipon** [comment]
@Graur please review this one

File: eo-runtime/src/main/java/org/eolang/Versionize.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import java.lang.annotation.Target;
+
+/**
+ * Annotation for an object which location should be extended with version package.
+ * More details <a href="https://github.com/objectionary/eo/issues/2506">here</a>
+ *
+ * @since 0.32.0
+ */
+@Target(ElementType.TYPE)
+@Versionize
```
Comment (Graur) [inline eo-runtime/src/main/java/org/eolang/Versionize.java:37]: @maxonfjvipon I'm not sure I understand why did you mark your annotation by your annotation?

File: eo-runtime/src/main/java/org/eolang/Versionize.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Target;
+
+/**
+ * Annotation for an object which location should be extended with version package.
+ * More details <a href="https://github.com/objectionary/eo/issues/2506">here</a>
+ *
+ * @since 0.32.0
+ */
+@Target(ElementType.TYPE)
```
Comment (Graur) [inline eo-runtime/src/main/java/org/eolang/Versionize.java:36]: @maxonfjvipon Maybe we can add a RetentionPolicy?

File: eo-runtime/src/main/java/org/eolang/Versionize.java (hunk 1 above)
Comment (maxonfjvipon) [reply eo-runtime/src/main/java/org/eolang/Versionize.java:37]: @Graur I just thought that maybe you're right and it's not necessary

File: eo-runtime/src/main/java/org/eolang/Versionize.java (hunk 2 above)
Comment (maxonfjvipon) [reply eo-runtime/src/main/java/org/eolang/Versionize.java:36]: @Graur sure

**Comment by maxonfjvipon** [comment]
@Graur please have a look one more time

**Comment by maxonfjvipon** [comment]
@Graur please have a look one more time

**Review by Graur — APPROVED** [review]

**Comment by yegor256** [comment]
@rultor merge

**Comment by yegor256** [comment]
@maxonfjvipon annotation is soft of a "quality" that we attribute to a every object of the given class. It can't be a verb. It must be a qualifier. Maybe "versionizable"? Or "VersionSensitive"? Or "Versionized"?

**Comment by maxonfjvipon** [comment]
@yegor256 please have a look

**Comment by maxonfjvipon** [comment]
@yegor256 please have a look

**Comment by yegor256** [comment]
@rultor merge

**Comment by yegor256** [comment]
@rultor merge

**Comment by yegor256** [comment]
@rultor status

**Pull request merged**